sources/httpUtils.ts

```diff
@@ -16,7 +16,7 @@ export async function fetchUrlStream(url: string, options: RequestOptions = {})
     const request = https.get(url, {...options, agent: proxyAgent}, response => {
       const statusCode = response.statusCode ?? 500;
```

Comment:
While we're at it, I think we should not use `500` for nullish statusCode, as it's likely going to confuse users more to have Corepack pretend it saw a 500 response code when it fact it didn't see anything.
```suggestion
      const statusCode = response.statusCode;
```

Reply:
Thanks for your spot. Updated.